client/state/form/reducer.js

```diff
@@ -14,6 +20,14 @@ reducers[ SET_FORM_PROPERTY ] = ( state, action ) => {
 	return Object.assign( {}, state, newObj );
 };
 
+reducers[ NEXT_STEP ] = ( state ) => {
+	return Object.assign( {}, state, { currentStep: state.currentStep + 1 } );
```

Comment:
To avoid explosions, should the reducer only advance when we aren't on the last step?